File: Classes/EVEOnlineAPI/API/EVECallList.m
//
//  EVECallList.m
//  EVEOnlineAPI
//
//  Created by Shimanski on 8/29/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "EVECallList.h"


@implementation EVECallListCallGroupsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"groupID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"name":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"callGroupDescription":@{@"type":@(EVEXMLSchemePropertyTypeString), @"elementName":@"description"}};
	
	return scheme;
}

@end


@implementation EVECallListCallsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"accessMask":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"type":@{@"type":@(EVEXMLSchemePropertyTypeScalar), @"transformer":^(id value) {
					   if ([value isKindOfClass:[NSString class]])
						return [value isEqualToString:@"Corporation"] ? @(EVECallTypeCorporation) : @(EVECallTypeCharacter);
					   else
						return @(EVECallTypeCharacter);
				   }},
				   @"name":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"groupID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"callDescription":@{@"type":@(EVEXMLSchemePropertyTypeString), @"elementName":@"description"}};
	
	return scheme;
}

@end


@implementation EVECallList

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"callGroups":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVECallListCallGroupsItem class]},
				   @"calls":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVECallListCallsItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVECallList.swift
//
//  EVECallList.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit


public class EVECallListCallGroup: EVEObject {
	public var name: String = ""
	public var callGroupDescription: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"name":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"callGroupDescription":EVESchemeElementType.String(elementName:"description", transformer:nil),
		]
	}
}

public class EVECallListCallsItem: EVEObject {
	public var accessMask: Int64 = 0
	public var type: EVECallType = .Character
	public var name: String = ""
	public var groupID: Int = 0
	public var callDescription: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"accessMask":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
		"type":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"name":EVESchemeElementType.String(elementName:nil, transformer:nil),
		"groupID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"callDescription":EVESchemeElementType.String(elementName:"description", transformer:nil),
		]
	}
}

public class EVECallList: EVEResult {
	public var callGroups: [EVECallListCallGroup] = []
	public var calls: [EVECallListCallsItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"callGroups":EVESchemeElementType.Rowset(elementName: nil, type: EVECallListCallGroup.self, transformer: nil),
			"calls":EVESchemeElementType.Rowset(elementName: nil, type: EVECallListCallsItem.self, transformer: nil)
		]
	}
}

